packages/store/addon/-private/ts-interfaces/minimum-adapter-interface.ts

```diff
@@ -0,0 +1,596 @@
+   * processing within the adapter.
+   *
+   * ⚠️ This Method is only called if `coalesceFindRequests` is `true`. The array passed to it is determined
+   * by the adapter's `groupRecordsForFindMany` method, and will be called once per group returned.
+   *
+   * It is often more effective to implement coalescing entirely inside your adapter instead
+   * of relying on this behavior from the store.
+   *
+   * See also `groupRecordsForFindMany` and `coalesceFindRequests`
+   *
+   * @method findMany [OPTIONAL]
+   * @public
+   * @optional
+   * @param {Store} store The store service that initiated the request being normalized
+   * @param {ModelSchema} schema An object with methods for accessing information about
+   *  the type, attributes and relationships of the primary type associated with the request.
+   * @param {Array<string>} ids An array of the ids of the resources to fetch
+   * @param {Array<Snapshot>} snapshots An array of snapshots of the available data for the resources to fetch
+   * @return {Promise} a promise resolving with resource data to feed to the associated serializer
+   */
+  findMany?(store: Store, schema: ModelSchema, ids: string[], snapshots: Snapshot[]): Promise<unknown>;
+
+  /**
+   * This method provides the ability to generate an ID to assign to a new record whenever `store.createRecord`
+   * is called if no `id` was provided.
+   *
+   * Alternatively you can pass an id into the call to `store.createRecord` directly.
+   *
+   * ```js
+   * let id = generateNewId(type);
+   * let newRecord = store.createRecord(type, { id });
+   * ```
+   *
+   * @method generateIdForRecord [OPTIONAL]
+   * @public
+   * @optional
+   * @param {Store} store The store service that initiated the request being normalized
+   * @param {String} type The type (or modelName) of record being created
+   * @param properties the properties passed as the second arg to `store.createRecord`
+   * @return {String} a string ID that should be unique (no other models of `type` in the cache should have this `id`)
+   */
+  generateIdForRecord?(store: Store, type: string, properties: unknown): string;
+
+  /**
+   * This method allows for you to split pending requests for records into multiple `findMany`
+   * requests. It receives an array of snapshots where each snapshot represents a unique record
+   * requested via `store.findRecord` during the most recent `runloop` that was not found in the
+   * cache or needs to be reloaded. It should return an array of groups.
+   *
+   * A group is an array of snapshots meant to be fetched together by a single `findMany` request.
+   *
+   * By default if this method is not implemented EmberData will call `findMany` once with all
+   * requested records as a single group.
+   *
+   * ⚠️ This Method is only called if `coalesceFindRequests` is `true`.
+   *
+   * ⚠️ It is often more effective to implement coalescing entirely inside your adapter instead
```

Comment:
@runspired we talked about giving people more info on when to do it and when not to